src/lib/merkle_map.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Experimental,
+  isReady,
+  shutdown,
+  Field,
+  Bool,
+  SelfProof,
+  CircuitValue,
+  arrayProp,
+  Poseidon,
+  prop,
+  Circuit,
+} from 'snarkyjs'
+
+import assert from 'assert';
+
+const bits = 255;
+
+const printDebugs = false;
+
+export class MerkleMap {
+  tree: InstanceType<typeof Experimental.MerkleTree>;
+
+  // ------------------------------------------------
+
+  constructor() {
+    assert(bits <= 255, 'bits must be <= 255')
+    if (bits != 255) {
+      console.warn('bits set to', bits + '. Should be set to 255 in production to avoid collisions');
+    }
+    this.tree = new Experimental.MerkleTree(bits+1);
+  }
+
+  // ------------------------------------------------
+
+  _keyToIndex(key: Field) {
+    // the bit map is reversed to make reconstructing the key during proving more convenient
+    let keyBits = key.toBits().slice(0,bits).reverse().map((b) => b.toBoolean());
+
+    var n = BigInt(0);
+    for (var i = 0; i < keyBits.length; i++) {
+      var b = keyBits[i] ? 1 : 0;
+      n += BigInt(2)**BigInt(i) * BigInt(b);
+    }
+
+    return BigInt(n)
```

Comment:
nit: `n` is already a bigint => `return n;`
another nit: would be nice to run this code through `prettier` (so we don't get another large formatting commit when someone with auto-formatting enabled makes a change to this file later)